cli/azd/pkg/project/project.go

```diff
@@ -271,7 +299,28 @@ func Save(ctx context.Context, projectConfig *ProjectConfig, projectFilePath str
 		return fmt.Errorf("saving project file: %w", err)
 	}
 
-	projectConfig.Path = projectFilePath
+	projectConfig.Path = filepath.Dir(projectFilePath)
```

Comment:
Just wondering - what drove this part of the change?

Reply:
I found an issue while I was testing running hooks defined in the infra folder.  The hooks were neither registered nor running. After debugging, I found that, for the `azd init` flow, the project is saved using the `projectFilePath`, which is pointing to the azure.yaml and not to its path:

![image](https://github.com/user-attachments/assets/c6d99ce1-d62b-4ce7-8cd0-6b2c0aa4e0e8)

W/o using `filepath.Dir(projectFilePath)`, the Path is saved using the azure.yaml in the path.

Then, during `project Load()`,  we were rewriting the path with the Dir, like:

![image](https://github.com/user-attachments/assets/cd708c9b-c4f4-43b7-80d1-c3102d5b744b)

In my bug, the hooks registration was happening After the first project `save()` and the first project `load()`, making the registration to use a wrong path as the project path.
However, after the first project Load(), the path was good as expected.

So, I removed the call to set the project path during the project `Load()` and make `Save()` to save it correctly since the beginning.

Does it makes sense?

Reply:
It does - thanks for the explanation here!